src/Soil/Biogeochemistry/Biogeochemistry.jl

```diff
@@ -137,6 +158,311 @@ function DETECTModelParameters(;
     )
 end
 
-include("./co2_parameterizations.jl")
+
+
+
+#################################
+
+"""
+    DETECTModel
+
+A model for simulating the production and transport of CO₂ in the soil with dynamic
+source and diffusion terms.
+"""
+struct DETECTModel{FT, PS, D, BC, S, DT} <: ClimaLSM.AbstractModel{FT}
```

Comment:
I wonder if we want to have an abstract type
``` abstract struct AbstractBiogeochemistryModel{FT} <: ClimaLSM.AbstractModel{FT} end```
?
and then `DETECTModel{FT, ...} <: AbstractBiogeochemistryModel{FT}`?

this would only really be necessary if you anticipate having multiple biogeochemistry models, though!

Reply:
I am doing it, but naming `SoilBiogeochemistry` to be more specific